packages/capacitor-plugin/src/web.ts

```diff
@@ -0,0 +1,284 @@
+import { WebPlugin, PluginListenerHandle } from '@capacitor/core';
+import type {
+  FileTransferPlugin,
+  DownloadFileOptions,
+  DownloadFileResult,
+  UploadFileOptions,
+  UploadFileResult,
+  ProgressStatus,
+  FileTransferError,
+} from './definitions';
+
+export class FileTransferWeb extends WebPlugin implements FileTransferPlugin {
+  private progressListeners: ((progress: ProgressStatus) => void)[] = [];
+  private lastProgressUpdate = 0;
+  private readonly PROGRESS_UPDATE_INTERVAL = 100; // 100ms between progress updates
+
+  async downloadFile(options: DownloadFileOptions): Promise<DownloadFileResult> {
+    try {
+      // First, download the file
+      const response = await this.makeRequest(options);
+      const blob = await response.blob();
+
+      // Create a URL for the blob
+      const url = URL.createObjectURL(blob);
+      
+      // Create a temporary anchor element
+      const a = document.createElement('a');
+      a.href = url;
+      a.download = this.extractFilename(options.path, options.url);
+      
+      // If progress is enabled, track the download
+      if (options.progress) {
+        const reader = new FileReader();
+        let loaded = 0;
+        const total = blob.size;
+
+        reader.onprogress = (event) => {
+          if (event.lengthComputable) {
+            loaded = event.loaded;
+            this.notifyProgress(options.url, loaded, total, true, 'download');
+          }
+        };
+
+        // Read the blob to track progress
+        reader.readAsArrayBuffer(blob);
+      }
+
+      // Trigger the download
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+
+      // Clean up the blob URL
+      URL.revokeObjectURL(url);
+
+      return {
+        blob,
+        path: options.path,
+      };
+    } catch (error) {
+      throw this.handleError(error, options.url, options.path);
+    }
+  }
+
+  async uploadFile(options: UploadFileOptions): Promise<UploadFileResult> {
+    try {
+      let blob: Blob;
+      if (options.blob) {
+        blob = options.blob;
+      } else {
+        // In web, we can't read from file path, so we need to throw an error
+        throw new Error('File upload from path is not supported in web. Please provide a blob instead.');
+      }
+
+      return new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest();
+        const url = this.buildUrl(options.url, options.params);
+
+        // Set up progress tracking if enabled
+        if (options.progress) {
+          xhr.upload.onprogress = (event) => {
+            if (event.lengthComputable) {
+              this.notifyProgress(options.url, event.loaded, event.total, true, 'upload');
+            }
```

Comment:
You could emit the event even if `lengthComputable` is false?

```suggestion
            this.notifyProgress(options.url, event.loaded, event.total, event.lengthComputable, 'upload');
```